src/test/java/it/smartcommunitylab/aac/saml/SamlIdentityProviderConfigurationTest.java

```diff
@@ -0,0 +1,320 @@
+
+        // Create (hardcoded) configuration map to be used for initialization
+        Map<String, Serializable> rawCfgMap = buildRawConfigurationMap(true);
+        SamlIdentityProviderConfigMap idProvCfgMap = new SamlIdentityProviderConfigMap();
+        idProvCfgMap.setConfiguration(rawCfgMap);
+
+        ConfigurableIdentityProvider cip = getDefaultConfigurableIdentityProvider();
+        IdentityProviderSettingsMap idProvSettingMap = getDefaultIdentityProviderSettingsMap();
+
+        SamlIdentityProviderConfig IdPCfg = new SamlIdentityProviderConfig(cip, idProvSettingMap, idProvCfgMap);
+        Assertions.assertThat(IdPCfg.getEntityId()).isEqualTo(expEntityID);
+    }
+
+    /**
+     * This test checks that a RelyingPartyRegistration contains information matching the saml configuration (with automatic idp metadata discovery)
+     * @throws Exception
+     */
+    @Test
+    public void validateRelyingPartyRegistrationAutomatic() throws Exception {
+
+        boolean automaticDiscoverySetting = true;
+        // Create (hardcoded) configuration map to be used for initialization
+        Map<String, Serializable> rawCfgMap = buildRawConfigurationMap(automaticDiscoverySetting);
+        SamlIdentityProviderConfigMap idProvCfgMap = new SamlIdentityProviderConfigMap();
+        idProvCfgMap.setConfiguration(rawCfgMap);
+
+        ConfigurableIdentityProvider cip = getDefaultConfigurableIdentityProvider();
+        IdentityProviderSettingsMap idProvSettingMap = getDefaultIdentityProviderSettingsMap();
+
+        SamlIdentityProviderConfig idProvCfg = new SamlIdentityProviderConfig(cip, idProvSettingMap, idProvCfgMap);
+        // Run mock server to locally catch the identity provider metadata request.
+        String IdpMetadata = fetchIdpMetadataExample();
+        WireMockServer mockIdpServer = buildMockIdpServer(IdpMetadata);
+        mockIdpServer.start();
+
+        // check that the obtained saml Identity Provider matches the obtained map
+        RelyingPartyRegistration rpRegistration = idProvCfg.getRelyingPartyRegistration();
+        mockIdpServer.stop();
+
+        Assertions.assertThat(rpRegistration.getEntityId()).isEqualTo(expEntityID);
+        Assertions.assertThat(rpRegistration.getRegistrationId()).isEqualTo(expProviderId);
+        Assertions.assertThat(rpRegistration.getSigningX509Credentials().size()).isEqualTo(1);
+        for (Saml2X509Credential credential : rpRegistration.getSigningX509Credentials()) {
+            // NOTE: these assertions currently assume that there is only one x509credential(s) -> for-loop is trivial
+            // NOTE: if there is more than one x509credential(s), iteration order would NOT be guaranteed as we are iterating on a Set
+
+            // check private information
+            PrivateKey obtPrivateKey = credential.getPrivateKey();
+            String obtPrivKeyb64Enc = Base64.getEncoder().encodeToString(obtPrivateKey.getEncoded());
```

Comment:
this reasoning is flawed. Keys and certificates are stored with PEM, a specific format. Encoding the key back to base64 is *not* guaranteed to produce the same string. 
If you want to check the key you need to actually **parse** the key yourself and then check if it matches the one parsed by the class you are testing. Same for cert.

Reply:
Parsed the key using [the linked spiring test](https://github.com/spring-projects/spring-security/blob/main/saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataRelyingPartyRegistrationConverterTests.java) as reference